client/src/Components/Map/index.tsx

```diff
@@ -0,0 +1,216 @@
+import BackgroundImage from './assets/TuckerMap.png'
+import AquaOcean from './assets/locations/AquaOcean.png'
+import BananaBunches from './assets/locations/BananaBunches.png'
+import BushLand from './assets/locations/BushLand.png'
+import CluckyCoop from './assets/locations/CluckyCoop.png'
+import CoolCloud from './assets/locations/CoolCloud.png'
+import DairyPark from './assets/locations/DairyPark.png'
+import FruityOrchard from './assets/locations/FruityOrchard.png'
+import GrainField from './assets/locations/GrainField.png'
+import GrazingLands from './assets/locations/GrazingLands.png'
+import HealthyTown from './assets/locations/HealthyTown.png'
+import SupplyStore from './assets/locations/SupplyStore.png'
+import VegieZone from './assets/locations/VegieZone.png'
+import WickedWaterway from './assets/locations/WickedWaterway.png'
+import YoghurtMountains from './assets/locations/YoghurtMountains.png'
+import ZombieWasteland from './assets/locations/ZombieWasteland.png'
+import SVGLocationGroup from './SVGLocationGroup'
+
+const Map: React.FC = () => {
+  const [height, setHeight] = useState(0)
+  const elementRef = useRef(null as null | HTMLDivElement)
+
+  const [selected, onSelect] = useState<Location | null>(null)
+
+  useEffect(() => {
+    if (elementRef?.current?.clientHeight) {
+      setHeight(elementRef?.current?.clientHeight)
+    }
+  }, []) //empty dependency array so it only runs once at render
+
+  const getClassname = (area: Location) =>
+    area === selected ? 'map-selected' : 'map-unselected'
+
+  const onMapClick = (area: Location) => {
+    console.log(area)
+    if (selected === area) {
+      onSelect(null)
+    } else {
+      onSelect(area)
+    }
+  }
+
+  // Data can be made from dev/svgParse.py
+  return (
+    <div
+      ref={elementRef}
+      className='flex-auto 2xl:flex-none xl:flex-none 2xl:h-4/5 xl:h-4/5 flex justify-center'
+    >
+      {height === 0 ? null : (
+        <div className='svgrow'>
+          <svg
+            xmlns='http://www.w3.org/2000/svg'
+            xmlnsXlink='http://www.w3.org/1999/xlink'
+            viewBox='0 0 1000 490.56'
+            height={height}
+            overflow='scroll'
+          >
+            <g id='bg'>
+              <image
+                width='7151'
+                height='3508'
+                transform='scale(0.14)'
+                xlinkHref={BackgroundImage}
+              />
+            </g>
+            <SVGLocationGroup
+              name={'coolCloud'}
+              width={1627}
+              height={300}
+              transform='translate(7) scale(0.589 0.603)'
+              image={CoolCloud}
+              className={getClassname('coolCloud')}
+              onClick={onMapClick}
+            />
+            <SVGLocationGroup
+              name={'fruityOrchard'}
+              width={1026}
+              height={743}
+              transform='translate(131 121) scale(0.139 0.148)'
```

Comment:
I probably wouldnt hardcode all these html elements instead map over a data array

Reply:
Okay I see the python script below. My comment is still relevant. Rather than generating the entire react component I would just generate an array which you can iterate over. This is a bit harder since you used python instead of node to write the script

Reply:
Ja, was just playing around, ezy fix, another 20 min